compiler/vm.nim

```diff
@@ -528,13 +528,51 @@ template maybeHandlePtr(node2: PNode, reg: TFullReg, isAssign2: bool): bool =
 when not defined(nimHasSinkInference):
   {.pragma: nosinks.}
 
+type
+  Profiler = object
+    tEnter: float
+    tos: PStackFrame
+
+proc dumpProfile(c: PCtx) =
+  if optProfileVM in c.config.globalOptions:
+    echo "VM profiler:"
+    var profile = c.profile
+    for i in 0..<32:
+      var tMax: float
+      var flMax: FileLine
+      for fl, t in profile:
+        if t > tMax:
+          tMax = t
+          flMax = fl
+
+      echo "  ", align($int(tMax * 1e6), 10), " µs: ", c.config.toMsgFilename(flMax.fileIndex), ":", flMax.line
```

Comment:
`miscdollars.toLocation`

Reply:
Not quite, I'm not using TLineInfo but my own object with linenum and fileindex. I don't remember why I did this though, I'll see if I can make it use the normal `TLineInfo` instead.

Reply:
asciitables doesn't left-align, does it?

Reply:
see tests in tests/compiler/tasciitables.nim, it auto-aligns based on tabs used as separators

Reply:
Sure, but adding a custom proc for left alignment results in more code then two straightforward `alignLeft()`s